lib/perl/Genome/Model/Tools/Htseq/Count.pm

```diff
@@ -71,260 +69,31 @@ class Genome::Model::Tools::Htseq::Count {
-        $final_bam = "$tmp/all_sequences.namesorted.cleaned.fixed.bam";
-        Genome::Sys->shellcmd(cmd => "$samtools_path fixmate $sorted_cleaned_bam $final_bam");
-    }
-
-    for my $type (qw/transcript gene/) {
-        $self->debug_message("Produce per-$type results...");
-        
-        # from Malachi's notes in JIRA issue TD-490
-        # samtools view -h chr22_tumor_nonstranded_sorted.bam | htseq-count --mode intersection-strict --stranded no --minaqual 1 --type exon --idattr transcript_id - chr22.gff > transcript_tumor_read_counts_table.tsv 
-   
-        my $cmd = $samtools_path 
-            . " view -h '$final_bam' "
-            . ($whitelist_alignments_flags ? " -f $whitelist_alignments_flags " : '')
-            . ($blacklist_alignments_flags ? " -F $blacklist_alignments_flags " : '')
-            . ' | '
-            . ($limit ? "head -n " . ($limit + $header_size) . " | " : '')
-            . $htseq_count_path
-            . " --mode $mode "
-            . " --minaqual $minaqual "
-            . " --stranded $htseq_stranded_param"
-            . " --type exon " # used for both gene and transcript iterations
-            . " --idattr ${type}_id"
-            . " -"
-            . " '$gtf_file'"
-            . " 1>'$output_dir/${type}-counts.tsv'"
-            . " 2>'$output_dir/${type}.err'";
-        
-        Genome::Sys->shellcmd(
-            cmd => $cmd, # "touch $output_dir/${type}-counts.tsv", #$cmd,
-            input_files => [$sorted_bam, $gtf_file],
-            #output_files => ["$output_dir/${type}-counts.tsv"],
-       );
-
-        my $method = $type . '_hits_file_path';
-        $self->$method("$output_dir/${type}-counts.tsv");
-    }
-
-    return 1;
-}
-
-sub _htseq_stranded_param_v2 {
-    my $self = shift;
-    my $transcript_strand = shift;
-
-    if ($transcript_strand eq 'unstranded') {
-        return 'no';
-    }
-    elsif ($transcript_strand eq 'firststrand') {
-        return 'reverse';
-    }
-    elsif ($transcript_strand eq 'secondstrand') {
-        return 'yes';
-    }
-    else {
-        die $self->error_message("Unknown transcript_strand $transcript_strand!  expected unstranded, firstread or secondread.");
-    }
-}
-
-sub _htseq_stranded_param_v1 {
-    my $self = shift;
-    my $transcript_strand = shift;
-
-    if ($transcript_strand eq 'unstranded') {
-        return 'no';
-    }
-    else {
-        die $self->error_message("Htseq::Count v1 only supports unstranded libraries, use v2 instead.");
-    }
-}
-
-sub _htseq_stranded_param {
-    my $self = shift;
-
-    if ($self->result_version == 1) {
-        $self->_htseq_stranded_param_v1(@_);
-    }
-    elsif ($self->result_version == 2) {
-        $self->_htseq_stranded_param_v2(@_);
-    }
-    else {
-        die "no _htseq_stranded_param implementation for version " . $self->result_version;
-    }
+sub result_class {
+    return __PACKAGE__ . '::Result';
 }
 
-sub _merge_v2 { _merge_v1(@_) }
-sub _merge_v1 {
+sub input_hash {
```

Comment:
That sounds good to me.  I'll merge this PR and open a new one that moves it up and removes other subclass implementations where they're not needed.